.github/workflows/main.yml

```
@@ -0,0 +1,52 @@
+name: CI
+on: 
+  pull_request:
+  push:
+    branches:
+      - main
+      - add_GHA_based_CI
+
+jobs:
+  make_sdist:
+    name: Make SDist
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - run: python -m pip install build
+
+      - run: python -m build .
+
+      - name: Upload SDist
+        uses: actions/upload-artifact@v4
+        with:
+          name: example
+          path: dist/*.tar.gz
+
+  build_wheels:
+    name: Wheel on ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+        
+      - run: python -m pip install build
+
+      - run: python -m build .
```

```suggestion
      - run: python -m build  --wheel
```

(`.` optional)

You are already building the SDist above, so no need to repeat it here.

---

Though admittedly, you only upload the wheel in the next line, and build will make the SDist and then build the wheel from it rather than directly making the wheel, so this might not be a good suggestion to take. If you had broken SDists your version would catch it, and mine wouldn't.

---

Just a warning (you did it correctly here, but in case it ever matters), GitHub Actions will actually corrupt the binaries if you download several artifacts with any matching files. It's an annoying bug, as you have to make sure not to upload SDists from more than one job or wheels that match.

Very annoying bug to track down. ;)